src/core/ledger/ledger.js

```diff
@@ -65,9 +75,30 @@ type MutableAccount = {|
   // may not receive or transfer Grain. Accounts start inactive, and must
   // be explicitly activated.
   active: boolean,
+  // key-value store of blockchain addresses the account receives grain to
+  payoutAddresses: PayableAddressStore,
 |};
 export type Account = $ReadOnly<MutableAccount>;
 
+/**
+ * The CurrencyId key must be stringified to ensure the data is retrievable.
+ * Keying on the raw CurrencyId object means keying on the object reference,
+ * rather than the contents of the object.
+ */
+type CurrencyIdString = string;
```

Comment:
It would be a lot more intuitive id `CurrencyId` was the string that's generated from a `Currency` object